Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnSelect.hs

```diff
@@ -108,6 +108,9 @@ buildSelectedQuote estimate providerInfo now QuoteInfo {..} = do
   uid <- generateGUID
   tripTerms <- buildTripTerms descriptions
   driverOffer <- buildDriverOffer estimate.id quoteDetails
+  searchRequest <-
```

Comment:
Instead of fetching searchRequest and then getting merchant id from it. Pass merchantId from where calling this function.